### ISSUE TITLE
Vulkan inactive ... not installed dependences ?

### ISSUE THREAD

**Issue by Augusto7743** [zaps166/QMPlay2#417]
Using the current version 21.03.9 64 in Ubuntu 21.04 with GT 640 current driver version.
QMPlayer2 need nvidia 32 libraries ?
I had used QMPlayer2 in another Ubuntu 21.04 with nvidia driver both 32 and 64 libraries installed and the Vulkan option was available.
Trying QMPlayer in system with nvidia driver installed only 64 bits libraries the option to choice Vulkan is displayed being inactive.

Here the log
[29 abr 2021 04:43:04.043] initInstance: No Vulkan library available
[29 abr 2021 04:43:04.052] Failed to create platform Vulkan instance
[29 abr 2021 04:43:04.052] Vulkan is unable to work with QMPlay2 on this platform: Can't create Vulkan instance: ErrorInitializationFailed
[29 abr 2021 04:46:05.839] Case insensitive sorting unsupported in the posix collation implementation
[29 abr 2021 04:46:05.839] Numeric mode unsupported in the posix collation implementation
[29 abr 2021 04:46:08.434] Case insensitive sorting unsupported in the posix collation implementation
[29 abr 2021 04:46:08.434] Numeric mode unsupported in the posix collation implementation
[29 abr 2021 04:46:09.840] Empty filename passed to function

QMPlayer even being 64 bits need also 32 libraries to run Vulkan ?
Thanks for reply.

**Comment by zaps166** [comment]
64bit software requires only 64bit libraries. How did you get QMPlay2? Can you give me your GPU model and driver version? Does other Vulkan software run properly?

**Comment by zaps166** [comment]
Hmm, The newest AppImage (QMPlay2 21.03.09) should use Vulkan properly on Ubuntu. I have to check it.

**Comment by zaps166** [comment]
Ubuntu 21.04, Nvidia proprietary drivers, `QMPlay2-21.03.09-1-x86_64.AppImage`, test result - success (except CUVID hardware acceleration):

![Screenshot from 2021-05-07 16-15-23](https://user-images.githubusercontent.com/6239090/117479182-898e1780-af4f-11eb-91e1-40fa0b7e7164.png)

---

Could you upload `vulkaninfo` result?
```
sudo apt install vulkan-tools
vulkaninfo > vulkaninfo.txt
```

Upload `vulkaninfo.txt` here.

**Comment by Augusto7743** [comment]
command line is displayed
ERROR: [Loader Message] Code 0 : /usr/lib/i386-linux-gnu/libvulkan_intel.so: wrong ELF class: ELFCLASS32
ERROR: [Loader Message] Code 0 : /usr/lib/i386-linux-gnu/libvulkan_lvp.so: wrong ELF class: ELFCLASS32
ERROR: [Loader Message] Code 0 : /usr/lib/i386-linux-gnu/libvulkan_radeon.so: wrong ELF class: ELFCLASS32
error: XDG_RUNTIME_DIR not set in the environment.
error: XDG_RUNTIME_DIR not set in the environment.
error: XDG_RUNTIME_DIR not set in the environment.
error: XDG_RUNTIME_DIR not set in the environment.
error: XDG_RUNTIME_DIR not set in the environment.
WARNING: lavapipe is not a conformant vulkan implementation, testing use only.
error: XDG_RUNTIME_DIR not set in the environment.
error: XDG_RUNTIME_DIR not set in the environment.
error: XDG_RUNTIME_DIR not set in the environment.
error: XDG_RUNTIME_DIR not set in the environment.
WARNING: lavapipe is not a conformant vulkan implementation, testing use only.

Installing using driver metapackage is installed some files with name reference to intel and radeon.
Not problem using vulkan in other softwares.

The apprun file from appimage has reference to vulkan
#!/bin/sh
QMPLAY2_DIR=`readlink -f $(dirname "$0")`
QMPLAY2_BINARY=$QMPLAY2_DIR/usr/bin/QMPlay2
ldd "$QMPLAY2_BINARY" 2>&1 | grep "libstdc++.so.6" | grep "CXXABI\|GLIBCXX\|not found" > /dev/null
if [ $? -eq 0 ]; then
    export LD_LIBRARY_PATH=$QMPLAY2_DIR/usr/lib/optional:$LD_LIBRARY_PATH
    echo "Using bundled libstdc++.so.6"
fi
if [ -z  ]; then
    if [ -e /usr/lib/x86_64-linux-gnu/libvulkan.so.1 ]; then
        export QT_VULKAN_LIB=/usr/lib/x86_64-linux-gnu/libvulkan.so.1
    fi
fi
exec "$QMPLAY2_BINARY" "$@"

I not understand if is command to load vulkan in qt.
Here is having problem if trying to install in opt folder all extracted files from appimage.

**Comment by zaps166** [comment]
> Here is having problem if trying to install in opt folder all extracted files from appimage.

Run AppImage if you test it, because something might be wrong if you start it other way than `AppRun` script.

---

Could you upload your `vulkaninfo` results?

---

I installed this: ![Screenshot_20210508_191354](https://user-images.githubusercontent.com/6239090/117547739-8a977580-b031-11eb-8c20-d8eee7c9e2ee.png)

**Comment by Augusto7743** [comment]
"Why it tries to load 32-bit libraries? Something must be wrong with your system configuration. As I wrote you - I installed Ubuntu 21.04 and everything is working properly (I also have 32-bit libs like /usr/lib/i386-linux-gnu/libvulkan_intel.so and no errors)."
When installing metapackage some dependencies also install libraries related with Intel and AMD even if only is used a Nvidia video card.
All installed correctly.
The error message is because the vulkaninfo try does an comand in intel and amd libraries and not having an intel and amd gpu result in error. Not exactly a problem.

QMPlay2 appimage apprun has details about command in loading vukan in qt. The path in command is correct, but look how if using that command can create problems.
Starting using appimage  is random if will or not use Vulkan.
Not has another way to avoid load vulkan librarie using apprun script ? Here I has another softwares using QT and Vulkan without problems and not having in apprun command about load vulkan.
If that problem is happening with anothers users perhaps they start and file and the playback runs thus not see having problems because automatically is used OpenGL.

Github is problem for some browsers ... not accept upload the txt file.
